### PR TITLE
Bump golang to v1.20 for ipam main branch prow tests

### PR DESCRIPTION
**Pull request by Sunnatillo**
As title

### CODE REVIEW

**Review by smoshiur1237 — COMMENTED** [review]
/lgtm

File: prow/manifests/overlays/metal3/config.yaml [hunk 1]
```diff
@@ -932,6 +932,21 @@ presubmits:
...
+        containers:
+          - args:
+              - ./hack/gomod.sh
+            command:
+              - sh
+            env:
+              - name: IS_CONTAINER
+                value: "TRUE"
+            image: docker.io/golang:1.20
+            imagePullPolicy: Always
```
Comment (Rozzii) [inline prow/manifests/overlays/metal3/config.yaml:947]: I think "Always" is a bit aggressive for imagePullPolicy, it think this should be pulled only if the image is not present

**Review by Rozzii — CHANGES_REQUESTED** [review]

File: prow/manifests/overlays/metal3/config.yaml (hunk 1 above)
Comment (Sunnatillo) [reply prow/manifests/overlays/metal3/config.yaml:947]: Noted. But there might be another reason, why we set `imagePullPolicy` to ` Always` for all images. I can test it in separately.

Comment (Sunnatillo) [reply prow/manifests/overlays/metal3/config.yaml]: changed to IfNotPresent. It will prevent downloading images for each prow test. Nice catch.

**Review by kashifest — APPROVED** [review]

File: prow/manifests/overlays/metal3/config.yaml [hunk 2]
```diff
@@ -1143,6 +1217,21 @@ presubmits:
     - name: generate
       branches:
         - main
+      skip_if_only_changed: '(((^|/)OWNERS)|(\.md))$'
```
Comment (mboukhalfa) [inline prow/manifests/overlays/metal3/config.yaml:1220]: cannot we filter all .md files (even README )

Comment (Sunnatillo) [reply prow/manifests/overlays/metal3/config.yaml:1220]: Can we take this as it is for now? I will handle this and `imagePullPolicy:IfNotPresent` in a separate PR

Comment (mboukhalfa) [reply prow/manifests/overlays/metal3/config.yaml:1220]: NP

**Review by Rozzii — APPROVED** [review]

**Pull request merged**